apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -176,23 +176,48 @@ const TutorialExplorer = React.createClass({
     }
   },
 
+  /**
+   * Given a sort by choice (popularityrank or displayweight) and a grade range,
+   * return the field name from the tutorials data that should used for sorting.
+   */
+  getSortByFieldName(sortBy, grade) {
+    let sortByFieldName;
+
+    // If we're sorting by recommendation (a.k.a. displayweight) then find the
+    // right set of data to match the currently-selected grade.
+    if (sortBy === TutorialsSortBy.displayweight) {
+      if (grade === "all") {
+        sortByFieldName = "displayweight";
+      } else if (grade === "pre" || grade === "2-5") {
+        sortByFieldName = "displayweight_k5";
+      } else if (grade === "6-8") {
+        sortByFieldName = "displayweight_middle";
+      } else {
+        sortByFieldName = "displayweight_high";
+      }
+    } else {
+      sortByFieldName = "popularityrank";
+    }
+
+    return sortByFieldName;
+  },
+
   /*
    * The main tutorial set is returned with the given filters and sort order.
    *
    * Whether en or non-en user, this filters as though the user is of "en-US" locale.
    */
   filterTutorialSet(filters, sortBy, orgName) {
+    const grade = filters["grade"][0];
```

Comment:
Tiny nit: `filters.grade[0]`.  Or better yet, can `filters` be an instance of a class that defines these as instance properties or methods?

```js
class Filters {
  constructor(grade, student_experience, activity_type, ...) {
    this.grade = grade;
    // ...
  }

  matchesFilter(tutorial) {
    // Logic from static filterTutorials.
  }
}
```

Reply:
Good idea to build a class, but given how deeply filters are used throughout the tutorialExplorer, I'd like to do that in a future PR.